src/thor/isochrone.cc

```diff
@@ -329,6 +333,16 @@ ExpansionRecommendation Isochrone::ShouldExpand(baldr::GraphReader& /*graphreade
   // prune the edge if its start is above max contour
   if (time > max_seconds_ && distance > max_meters_)
     return ExpansionRecommendation::prune_expansion;
+
+  // log it back to expansion_action
+  if (expansion_callback_) {
+    if (time <= (max_seconds_ - METRIC_PADDING * kSecondsPerMinute) ||
+        distance <= (max_meters_ - METRIC_PADDING * kMetersPerKm)) {
+      expansion_callback_(graphreader, pred.edgeid(), "dijkstras", "s", pred.cost().secs,
```

Comment:
i'm trying to think of a cleaner way to do this that doesnt involve moving this out of the base class. it would be awesome if here we could simply set a member variable boolean, to false or something that would cause it not to do anything when we detect we dont want to actually call it back.

actually i guess you could keep an empty callback as a memeber variable and swap it in and out based on whether or not we are in the "extra" part of the expansion or not.

do you have any thoughts on that?

Reply:
sure no problem, I'll see what makes sense in order to not remove it in the base dijkstra class

Reply:
just wanted to get to this real quick, only to (re-)discover the initial problem: `max_meters_` and `max_seconds_` are member vars of `Isochrone` not `Dijkstras` and they're set during constructing the `isotile_`. that's why I moved the expansion_callback call in the first place.

in the light of that, I'm not sure I understand how your previous suggestions would be implemented.. we'd somehow have to let dijkstras know about the metrics if the call should stay where it was, and the metrics might differ for other dijkstra derived algos. also the expansion_callback is universal for all algos, so we can't just plug in algo-specific args into the lambda easily. 

right now I have no idea how to keep the function call in the base class.. do you?